src/tester.rs

```diff
@@ -439,15 +343,162 @@ fn write_with_colour(s: &str, colour: Color) {
     stderr.reset().ok();
 }
 
-fn output_failed() {
-    write_with_colour("FAILED", Color::Red);
+fn usage() -> ! {
+    eprintln!("Usage: <filter1> [... <filtern>]");
+    process::exit(1);
 }
 
-fn output_ok() {
-    write_with_colour("ok", Color::Green);
+/// Check for the case where the user has a test called `X` but `test_cmds` doesn't have a command
+/// with a matching name. This is almost certainly a bug, in the sense that the test can never,
+/// ever fire.
+fn check_names(cmd_pairs: &Vec<(String, Command)>, tests: &HashMap<String, Test>) {
+    let cmd_names = cmd_pairs.iter().map(|x| &x.0).collect::<HashSet<_>>();
+    let test_names = tests.keys().map(|x| x).collect::<HashSet<_>>();
+    let diff = test_names
+        .difference(&cmd_names)
+        .map(|x| x.as_str())
+        .collect::<Vec<_>>();
+    if !diff.is_empty() {
+        panic!(
+            "Command name(s) '{}' in tests are not found in the actual commands.",
+            diff.join(", ")
+        );
+    }
 }
 
-fn usage() -> ! {
-    eprintln!("Usage: <filter1> [... <filtern>]");
-    process::exit(1);
+/// Run every test in `test_files`, returning a tuple `(failures, num_ignored)`.
+fn run_tests(
+    test_files: Vec<PathBuf>,
+    inner: Arc<LangTesterInner>,
+) -> (Vec<(String, TestFailure)>, usize) {
+    let failures = Arc::new(Mutex::new(Vec::new()));
+    let mut num_ignored = 0;
+    let pool = ThreadPool::new(num_cpus::get());
+    'b: for p in test_files {
```

Comment:
Shall we remove the loop label if it's not used? (I'm also really not a fan of how it looks like a lifetime!)

Reply:
Good point!